### ISSUE TITLE
Ability to add a group of users to a channel

### ISSUE THREAD

**Issue by geekgonecrazy** [RocketChat/Rocket.Chat#3350]
If you have say a sales team, dev team etc.  Being able to add an entire group or team to a channel would be very useful

**Comment by skizzors** [comment]
Agreed! Or the ability to add all users to a channel! Thats VERY important

**Comment by krzysztofspilka** [comment]
Also agree, desperately need that one.

**Comment by engelgabriel** [comment]
Depends on #3522

**Comment by pablocerdeira** [comment]
There is a "temporary solution" for this. Create some read-only channels with the necessary members, then use, on the other channel, the command /invite-all-from <channel>.
Example:
1. Create a read-only channel called team-sales
2. Add all team sales users to this channel
3. Finally, go to all the channels you would like to include the sales team (ie: opportunities, sales-meetings etc) and add all users from you fake channel team-sales with: /invite-all-from team-sales

Maybe a much simple solution for the Rocket.Chat developers would create a hidden channel type (teams) to allow this team organization without deep coding.

**Comment by sampaiodiego** [comment]
is the slash commands `/invite-all-from` and `/invite-all-to` enough?

**Comment by geekgonecrazy** [comment]
I think this ties into the groups or teams functionality like github has with `@rocketchat/core`

**Comment by maudrid** [comment]
Hi. My employer is interested in developing this feature, but we would like to know if there are any requirements that you would like us to consider. This is so that we have the best chance of having a pull request accepted. This feature is blocking requirement for us if we want continue using RocketChat, so we have to build it or find a different solution.

**Comment by Sing-Li** [comment]
@maudrid   thanks for reaching out.  Yes, it is definitely on the roadmap and we should have a discussion.   Please email me at  sing dot li at rocket dot chat   and continue this conversation with the related team members.

**Comment by maudrid** [comment]
Thanks, I will find a time slot and let you know on Monday.

**Comment by Sing-Li** [comment]
@maudrid  Please have your team involved go through [this 5 years old discussion](https://forums.rocket.chat/t/ideas-on-multi-tenant-rocket-chat-support/2979)  just in case your intention of "scope of this feature" go beyond what is mentioned ONLY ON THIS ticket (shortcut to add a group of users to a channel).  Thanks.

**Comment by engelgabriel** [comment]
Hi, @maudrid and @Sing-Li I'd like to participate in this discussion. I don't think this is much related to multi-tenant. I think it is more related to how MS Teams work. One team is not necessarily completely isolated from the other team, but they can have access to different channels and me mentioned to as a single entity, etc.

**Comment by maudrid** [comment]
I would like to add that we would not be able to start development on this
immediately, but I have to start the planning as part of our road map.

**Comment by maudrid** [comment]
I have reviewed the other issue. Our purpose is just to make it easy to add/remove a pre-selected list of users to a channel without having to manage individual users.
Our current work around is to have an external list of users and adding and removing users from channels using the API, the problem is keeping all of this in sync is a nightmare.

**Comment by maudrid** [comment]
@Sing-Li I have sent you a few emails but have not gotten a reply. Am I using the correct email address?

**Comment by Sing-Li** [comment]
I did not see them @maudrid    please try again sing dot li at rocket dot chat.     Or find me on https;//open.rocket.chat

**Comment by rasos** [comment]
Teams are now available since RC 3.13. You can assign channels to a team. See https://docs.rocket.chat/guides/user-guides/teams/teams-channels
